cmpserver/plugin/plugin.go

```diff
@@ -297,11 +297,11 @@ func (s *Service) matchRepositoryGeneric(stream MatchRepositoryStream) error {
 		return fmt.Errorf("match repository error receiving stream: %w", err)
 	}
 
-	isSupported, isDiscoveryEnabled, err := s.matchRepository(bufferedCtx, workDir, metadata.GetEnv(), metadata.GetAppRelPath())
```

Comment:
This is nice! I felt `isDiscoveryEnabled` and `cfg.IsDiscoveryConfigured` were redundant upon reading these code the first time.